src/sentry/api/endpoints/sentry_apps.py

```diff
@@ -0,0 +1,36 @@
+from __future__ import absolute_import
+
+from rest_framework.permissions import IsAuthenticated
+from rest_framework.response import Response
+
+from sentry import features
+from sentry.api.base import Endpoint, SessionAuthentication
+from sentry.api.paginator import OffsetPaginator
+from sentry.api.serializers import serialize
+from sentry.constants import SentryAppStatus
+from sentry.models import SentryApp
+
+
+class SentryAppsEndpoint(Endpoint):
+    authentication_classes = (SessionAuthentication, )
+    permission_classes = (IsAuthenticated, )
+
+    def get(self, request):
+        if not any(features.has('organizations:internal-catchall', org)
+                   for org in request.user.get_orgs()):
+            return Response(status=404)
+
+        if request.user.is_superuser:
+            # Superusers have access to all apps, published and unpublished
+            queryset = SentryApp.objects.all()
+        else:
+            # Anyone else only has access to published apps
+            queryset = SentryApp.objects.filter(status=SentryAppStatus.PUBLISHED)
```

Comment:
Should normal users only be able to see their apps, or is this the public listing?

Reply:
@markstory yeah for now it's a effectively a public listing. Going forward will figure out how to show people _their own_ unpublished apps.